### PR TITLE
[MRG] Doc update.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3170
Closes https://github.com/mne-tools/mne-python/issues/3168
Also removes the boxes from manual index page.

Edit:
Closes https://github.com/mne-tools/mne-python/issues/3179
Closes https://github.com/mne-tools/mne-python/issues/3180

### CODE REVIEW

**Comment by Eric89GXL** [comment]
LGTM, here is the changed manual page for anyone else who is curious:

https://circle-artifacts.com/gh/jaeilepp/mne-python/801/artifacts/0/home/ubuntu/mne-python/doc/_build/html/manual/index.html

I'll run the ICA example in a bit.

**Comment by jaeilepp** [comment]
And the question in https://github.com/mne-tools/mne-python/issues/3170 is solved here by just adding ``TimeDecoding`` to it's place. Not sure what we should do to the API reference in the long run...

**Comment by Eric89GXL** [comment]
In `plot_artifacts_correction_ica.py`:
```
/home/larsoner/custombuilds/mne-python/mne/utils.py:558: DeprecationWarning: Function index_as_time is deprecated; index_as_time is deprecated and will be removed in 0.13, use raw.times[idx] (or raw.times[idx + raw.first_samp] instead
```
Otherwise it runs well

**Comment by Eric89GXL** [comment]
Looks like there are only a few instances if you want to get them all:

tutorials/plot_artifacts_correction_ica.py:start, stop = raw.index_as_time([raw.first_samp, raw.last_samp])
tutorials/plot_artifacts_correction_rejection.py:onset = raw.index_as_time(eog_events[:, 0]) - 0.25
tutorials/plot_brainstorm_auditory.py:onsets = raw.index_as_time(annotations_df['onset'].values)
tutorials/plot_brainstorm_auditory.py:durations = raw.index_as_time(annotations_df['duration'].values)

Can also be a separate PR if you want.

**Comment by jasmainak** [comment]
any reason why you removed the boxes? The reason I had them there was that it was easier to digest if it was all in one page rather than scrolling down 2 pages ...

**Comment by jasmainak** [comment]
Also, the images don't link here properly: https://circle-artifacts.com/gh/jaeilepp/mne-python/801/artifacts/0/home/ubuntu/mne-python/doc/_build/html/manual/decoding.html

**Comment by Eric89GXL** [comment]
The PR builds don't run examples so not linking to the sphinx-gallery generated images is to be expected

**Comment by jaeilepp** [comment]
> any reason why you removed the boxes? 

@agramfort didn't like them. They were too similar to tutorials page.

**Comment by jona-sassenhagen** [comment]
@jaeilepp is there a picture somewhere? It's probably fine, but I'd like to see ...

**Comment by jona-sassenhagen** [comment]
Of the ICA stuff.

**Comment by jona-sassenhagen** [comment]
I liked the boxes btw ... Now the layout is horizontally inefficient.

**Comment by Eric89GXL** [comment]
I slightly prefer this way for the Manual part, I find it easier to read as a table of contents. But I'm fine either way

**Comment by jona-sassenhagen** [comment]
>I'm fine either way

Yes.

**Comment by jaeilepp** [comment]
> @jaeilepp is there a picture somewhere? It's probably fine, but I'd like to see ...

Well not yet. I think the easiest way is to pull this and make the docs with ``html_dev-pattern``.

**Comment by agramfort** [comment]
+1 for removing the boxes in manual

@Eric89GXL regarding index_as_time I realized it's not as simple as doing raw.times[idx] as you need to take care of the first_samp :( so the method is to me still useful.

Maybe having raw.times[0] == 0. for raw and not first_samp / sfreq is mistake/bug?

**Comment by Eric89GXL** [comment]
In the deprecation message we say to just do `raw.times[idx] + raw.first_samp`. Too hard for users?

Personally I think that having `first_samp` at all is pretty frustrating, it's not involved for any of the other readers and we hit bugs related to it frequently. I'm not sure what value it provides other than to be backward compatible.

**Comment by Eric89GXL** [comment]
or I mean `raw.times[idx - raw.first_samp]` if `idx` has the offset

**Comment by agramfort** [comment]
ok can you try a PR to fix and see how the diff looks like?

**Comment by jaeilepp** [comment]
So I added a troubleshooting section to install page:
https://circle-artifacts.com/gh/jaeilepp/mne-python/805/artifacts/0/home/ubuntu/mne-python/doc/_build/html/install_mne_python.html#troubleshooting
See if it's good enough. I'm not sure why the error in https://github.com/mne-tools/mne-python/issues/3179 occurred.

File: doc/install_mne_python.rst [hunk 1]
```diff
@@ -250,3 +250,32 @@ our visualization capabilities. Please refer to the
...
+
+.. code-block:: bash
+
+    >>> import os
+    >>> os.environ['ETS_TOOLKIT'] = 'qt4'
+    >>> os.environ['QT_API'] = 'pyqt'
+
+This will tell Traits that we will use Qt with PyQt bindings.
+
+If you get an error saying:
```
Comment (Eric89GXL) [inline doc/install_mne_python.rst]: double-colon (for code rendering)

File: doc/install_mne_python.rst [hunk 2]
```diff
@@ -250,3 +250,32 @@ our visualization capabilities. Please refer to the
 `PySurfer installation page <https://pysurfer.github.io/install.html>`_
 for up to date information.
 
+Troubleshooting
+###############
+
+If you run into trouble when visualizing source estimates (or anything else
+using mayavi), you can try setting ETS_TOOLKIT environment variable.
+
+.. code-block:: bash
```
Comment (Eric89GXL) [inline doc/install_mne_python.rst]: this is not bash

File: doc/install_mne_python.rst [hunk 3]
```diff
@@ -250,3 +250,32 @@ our visualization capabilities. Please refer to the
...
+
+    >>> import os
+    >>> os.environ['ETS_TOOLKIT'] = 'qt4'
+    >>> os.environ['QT_API'] = 'pyqt'
```
Comment (Eric89GXL) [inline doc/install_mne_python.rst]: you can also give the alternative bash Python-launching version, something like
```
.. code-block:: bash

    $ ETS_TOOLKIT=qt4 QT_API=pyqt ipython

or:

.. code-block:: bash

    $ export ETS_TOOLKIT=qt4
    $ export QT_API=pytq
    $ ipython
```

File: doc/install_mne_python.rst [hunk 4]
```diff
@@ -250,3 +250,32 @@ our visualization capabilities. Please refer to the
...
+This will tell Traits that we will use Qt with PyQt bindings.
+
+If you get an error saying:
+
+    ValueError: API 'QDate' has already been set to version 1
+
+you have run into a conflict with Traits. You can work around this by telling
+the interpreter to use QtGui and QtCore from pyface.
+
+.. code-block:: bash
```
Comment (Eric89GXL) [inline doc/install_mne_python.rst]: also not bash

Comment (agramfort) [inline doc/install_mne_python.rst]: eric merge when you're happy

**Comment by jaeilepp** [comment]
Okay, I added the inclusion of ref_meg channels to ``raw.plot``. It was causing  failures on circle.
Ready for review.

**Comment by Eric89GXL** [comment]
Awesome, 4 issues in one PR. Thanks @jaeilepp

**Comment by jasmainak** [comment]
@jaeilepp on fire before leaving for vacation :)

**Comment by jona-sassenhagen** [comment]
Thanks for fixing the ICA bug. 





On Thu, Apr 28, 2016 at 7:15 AM -0700, "Mainak Jas" <notifications@github.com> wrote:












@jaeilepp on fire before leaving for vacation :)

**Pull request merged**